sql-plugin/src/main/scala/org/apache/spark/sql/rapids/execution/ShuffledBatchRDD.scala

```diff
@@ -49,10 +52,10 @@ class BatchPartitionIdPassthrough(override val numPartitions: Int) extends Parti
 class CoalescedBatchPartitioner(val parent: Partitioner, val partitionStartIndices: Array[Int])
   extends Partitioner {
 
-  @transient private lazy val parentPartitionMapping = {
+  @transient private lazy val parentPartitionMapping: Array[Int] = {
     val n = parent.numPartitions
     val result = new Array[Int](n)
-    for (i <- partitionStartIndices.indices) {
+    for (i <- 0 until partitionStartIndices.length) {
```

Comment:
Just a note. There are several changes here, like this one, that make it look like the code is going backwards based off of the IDEs warnings.  This is because I made the code match as closely as possible to the Spark 3.2.0-RC2 code. This was so that differences between it and other versions of the code are much more obvious.